dapps/profile.vue/src/components/did/DidService.ts

```diff
@@ -40,45 +40,68 @@ export class DidService {
    * Returns the delegates excluding the owner entry
    * @param didDocument specified did document
    */
-  // eslint-disable-next-line class-methods-use-this
-  async getDelegates(didDocument: DidDocument): Promise<Delegate[]> {
-    // regex to remove #key suffixes
-    const regex = /([#])(key-)(\d)+/;
-
-    return Promise.all(didDocument.authentication
-      // Filter out owner
-      .filter((auth) => auth !== DidService.getOwner(didDocument))
-      .map((auth) => auth.replace(regex, ''))
-      .map(async (did) => ({
-        did,
-        // Currently not using the note
-        // note: await profileUtils.getUserAlias(this.runtime, await this.runtime.did.convertDidToIdentity(did)),
-      })));
+  static getDelegates(didDocument: DidDocument): Delegate[] {
+    const keyIds = didDocument.publicKey.map((key) => key.id);
+
+    return didDocument.authentication
+      .filter((auth) => keyIds.includes(auth))
+      .map((key) => didDocument.publicKey.find((entry) => entry.id === key))
+      // Filter out "owner"
+      .filter((delegate) => delegate.controller !== didDocument.id);
   }
 
-  static updateDelegates(didDoc: DidDocument, delegates: Delegate[]): DidDocument {
-    // Preserve the owner
-    const ownerEntry = DidService.getOwner(didDoc);
-    const newDoc = didDoc;
-    newDoc.authentication = [
-      ownerEntry,
-      ...delegates.map((delegate) => delegate.did),
-    ];
-    return newDoc;
+  /**
+   * Enhances a DID Document with given delegates
+   * @param newDelegates Array of DIDs to be added as delegate
+   * @param didDoc DID Document to be enhanced
+   */
+  async updateDelegates(newDelegates: string[], didDoc: DidDocument): Promise<DidDocument> {
+    const delegateKeys = DidService.getDelegates(didDoc).map((delegate) => delegate.id);
+    const type = 'Secp256k1VerificationKey2018';
+
+    // DidDoc without delegates
+    const cleanDidDoc = {
+      ...didDoc,
+      publicKey: didDoc.publicKey.filter((key) => !delegateKeys.includes(key.id)),
```

Comment:
isn't pulicKey and authentication overwritten anyways with the next step?

Reply:
No, because the cleaned values are spreaded into the new object. The new delegates are added on top :)